src/pages/TermsOfService.test.ts

```diff
@@ -0,0 +1,228 @@
+    },
+  };
+});
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock(
+  'src/libs/nav',
+  (): NavExports => ({
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    goToPath: jest.fn(),
+  })
+);
+type AuthExports = typeof import('src/libs/auth');
+jest.mock(
+  'src/libs/auth',
+  (): AuthExports => ({
+    ...jest.requireActual('src/libs/auth'),
+    signOut: jest.fn(),
+  })
+);
+
+const setupMockAjax = (termsOfService) => {
+  const getTos = jest.fn().mockResolvedValue('some text');
+  const getTermsOfServiceComplianceStatus = jest.fn().mockResolvedValue(termsOfService);
+  const getStatus = jest.fn().mockResolvedValue({});
+  const acceptTos = jest.fn().mockResolvedValue({
+    userInfo: {
+      userSubjectId: 'testSubjectId',
+      userEmail: 'testEmail',
+    },
+    enabled: {
+      ldap: true,
+      allUsersGroup: true,
+      google: true,
+    },
+  });
+  const rejectTos = jest.fn().mockResolvedValue({
+    userInfo: {
+      userSubjectId: 'testSubjectId',
+      userEmail: 'testEmail',
+    },
+    enabled: {
+      ldap: true,
+      allUsersGroup: true,
+      google: true,
+    },
+  });
+  const getFenceStatus = jest.fn();
+  const getNihStatus = jest.fn();
+
+  type AjaxContract = ReturnType<typeof Ajax>;
+  type UserContract = ReturnType<typeof User>;
+  type MetricsContract = ReturnType<typeof Metrics>;
+  type GroupsContract = ReturnType<typeof Groups>;
+
+  asMockedFn(Ajax).mockImplementation(
+    () =>
+      ({
+        Metrics: {
+          captureEvent: jest.fn(),
+        } as Partial<MetricsContract>,
+        User: {
+          profile: {
+            get: jest.fn().mockResolvedValue({ keyValuePairs: [] }),
+            set: jest.fn().mockResolvedValue({ keyValuePairs: [] }),
+            setPreferences: jest.fn().mockResolvedValue({}),
+            preferLegacyFirecloud: jest.fn().mockResolvedValue({}),
+          },
+          getTos,
+          getTermsOfServiceComplianceStatus,
+          getStatus,
+          acceptTos,
+          rejectTos,
+          getFenceStatus,
+          getNihStatus,
+        } as Partial<UserContract>,
+        Groups: {
+          list: jest.fn(),
+        } as Partial<GroupsContract>,
+      } as Partial<AjaxContract> as AjaxContract)
+  );
+
+  const signInStatus = 'signedIn';
+  authStore.update((state) => ({ ...state, termsOfService, signInStatus }));
+  return {
+    getTosFn: getTos,
+    getStatusFn: getStatus,
+    getTermsOfServiceComplianceStatusFn: getTermsOfServiceComplianceStatus,
+    acceptTosFn: acceptTos,
+    rejectTosFn: rejectTos,
+  };
+};
+
+describe('TermsOfService', () => {
+  it('fetches the Terms of Service text from Sam', async () => {
+    // Arrange
+    const termsOfService = {
+      userHasAcceptedLatestTos: true,
+      permitsSystemUsage: true,
+    };
+
+    const { getTosFn } = setupMockAjax(termsOfService);
+
+    // Act
+    await act(async () => {
+      render(h(TermsOfServicePage));
+    });
```

Comment:
Since this is repeated in every test, it might be worth moving into the `setup` function.

Reply:
I actually think it could be confusing to move it there since that fn is really just focused around setting up mocks. I'd like to just keep the page renders where they are to make it clear what the order of the test is.